src/surveys/surveys.controller.ts

```diff
@@ -1,4 +1,118 @@
-import { Controller } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  Get,
+  HttpStatus,
+  Param,
+  Patch,
+  Post,
+  Res,
+  UseGuards,
+} from '@nestjs/common';
+import { SurveyService } from './surveys.service';
+import { ApiResponse, ApiTags, ApiParam } from '@nestjs/swagger';
+import { CreateSurveyDTO, UpdateSurveyDTO } from 'src/surveys/surveys.dto';
+import { AdminRolesGuard } from 'src/auth/authGuard';
 
-@Controller('surveys')
-export class SurveysController {}
+@Controller('survey')
+@ApiTags('Survey')
+export class SurveysController {
+  constructor(private surveyService: SurveyService) {}
+
+  @ApiResponse({
+    status: 200,
+    description: 'Survey has been succesfully created',
+  })
+  // @UseGuards()
+  @Post('/create')
+  async createSurvey(
+    @Res() response,
+    @Body()
+    body: CreateSurveyDTO,
+  ) {
+    try {
+      const newSurvey = await this.surveyService.createSurvey(body);
+
+      return response.status(HttpStatus.CREATED).json({
```

Comment:
You need to standardize your response across board for both a successful response and a failed response. So each of the for example should have {statusCode, message, data(which is an object), either a success or error field which is a boolean- you can add both}

Reply:
noted boss